test/unit/org/apache/cassandra/index/sai/cql/AllowFilteringTest.java

```diff
@@ -296,6 +296,91 @@ public void testAllowFilteringOnClusteringAndRegularColumns() throws Throwable
         test("SELECT * FROM %s WHERE v1=0 AND v2=0 AND k1=0 AND k2=0 AND (c1, c2, c3, c4) = (0, 0, 0, 0) AND v3=0", true);
     }
 
+    @Test
+    public void testAllowFilteringTextWithINClause ()
+    {
+        createTable("CREATE TABLE %S (k1 TEXT, k2 TEXT, k3 TEXT, PRIMARY KEY(k1))");
+        createIndex("CREATE INDEX ON %s(K2) USING 'sai'");
+
+        execute("INSERT INTO %s (k1,k2,k3) VALUES ('s1','s11','s111')");
+        execute("INSERT INTO %s (k1,k2,k3) VALUES ('s2','s11','s11')");
+        execute("INSERT INTO %s (k1,k2,k3) VALUES ('s3','s22','s111')");
+        execute("INSERT INTO %s (k1,k2,k3) VALUES ('s4','s22','s111')");
+        execute("INSERT INTO %s (k1,k2,k3) VALUES ('s5','s31','s111')");
+
+        assertRowCount(execute("SELECT * FROM %s WHERE k2='s11' AND k3 IN ('s11','s111') ALLOW FILTERING"),2);
+        assertRowCount(execute("SELECT * FROM %s WHERE k2='s22' AND k3 IN ('s111','s111') ALLOW FILTERING"), 2);
+        assertRowCount(execute("SELECT * FROM %s WHERE k2='s22' AND k3 IN ('s','s1') ALLOW FILTERING"), 0);
+        // To test if an IN clause without an AND condition does not create a query plan and works as expected.
+        assertRowCount(execute("SELECT * FROM %s WHERE k2 IN ('s11','s22') ALLOW FILTERING"), 4);
```

Comment:
I noticed that we return if we remove the ALLOW FILTERING:
` Cannot execute this query as it might involve data filtering and thus may have unpredictable performance. If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING`
Shouldn't we improve the error to explain that IN is not supported with SAI indexes?
I find this confusing for an end user. In addition to the docs that say that SAI supports it :( 

Maybe in a follow up ticket or so?

Reply:
Generally making the "why" in cases where AF is required more obvious could be a follow-up for sure. (At the end of the day, there's a part of me that would like to just get rid of AF as a thing, but yeah...I know it can still be dangerous.)

@sunil9977 Let's at least fix the docs here, removing both `OR` and `IN` form the supported operator list in `doc/modules/cassandra/partials/sai/supported-query-operators-list.adoc`. This is the doc you're referring to @ekaterinadimitrova2?